packages/taquito/test/contract/rpc-contract-provider.spec.ts

```diff
@@ -1856,4 +1856,233 @@ describe('RpcContractProvider test', () => {
       done();
     });
   });
+
+  describe('getSaplingDiffByID', () => {
+    it('should call getSaplingDiffById', async (done) => {
+      mockRpcClient.getBlock.mockResolvedValue({ header: { level: 123456 } });
+      mockRpcClient.getSaplingDiffById.mockResolvedValue({
+        root: 'fbc2f4300c01f0b7820d00e3347c8da4ee614674376cbc45359daa54f9b5493e',
+        commitments_and_ciphertexts: [],
+        nullifiers: [],
+      });
+
+      const result = await rpcContractProvider.getSaplingDiffByID('133');
+      expect(result.root).toEqual(
+        'fbc2f4300c01f0b7820d00e3347c8da4ee614674376cbc45359daa54f9b5493e'
+      );
+      expect(result.commitments_and_ciphertexts).toEqual([]);
+      expect(result.nullifiers).toEqual([]);
+
+      expect(mockRpcClient.getSaplingDiffById.mock.calls[0][0]).toEqual('133');
+      expect(mockRpcClient.getSaplingDiffById.mock.calls[0][1]).toEqual({ block: 'head' }); // no block specified
+      done();
+    });
+
+    it('should call getSaplingDiffById with a specified block level', async (done) => {
+      mockRpcClient.getBlock.mockResolvedValue({ header: { level: 123456 } });
+      mockRpcClient.getSaplingDiffById.mockResolvedValue({
+        root: 'fbc2f4300c01f0b7820d00e3347c8da4ee614674376cbc45359daa54f9b5493e',
+        commitments_and_ciphertexts: [],
+        nullifiers: [],
+      });
+
+      const result = await rpcContractProvider.getSaplingDiffByID('133', 654321);
+      expect(result.root).toEqual(
+        'fbc2f4300c01f0b7820d00e3347c8da4ee614674376cbc45359daa54f9b5493e'
+      );
+      expect(result.commitments_and_ciphertexts).toEqual([]);
+      expect(result.nullifiers).toEqual([]);
+
+      expect(mockRpcClient.getSaplingDiffById.mock.calls[0][0]).toEqual('133');
+      expect(mockRpcClient.getSaplingDiffById.mock.calls[0][1]).toEqual({ block: '654321' });
+      done();
+    });
+  });
+
+  describe('Testing Bug #1762 mapTypecheckError', () => {
+    it('should have a defined response and storage of 3 should not throw an error', async (done) => {
+      mockRpcClient.getEntrypoints.mockResolvedValue({
+        entrypoints: {},
+      })
+      mockRpcClient.getContract.mockResolvedValue({
```

Comment:
This mocked value takes considerable space in the test. Consider adding it to a separate file and importing it from there.

Reply:
👍